src/controllers/accounts.js

```diff
@@ -75,32 +48,20 @@ exports.getResource = function * fetch () {
  *
  * @return {void}
  */
-exports.putResource = function * putResource () {
-  const self = this
-  let name = this.params.name
+function * putResource () {
+  const name = this.params.name
   request.validateUriParameter('name', name, 'Identifier')
-  name = name.toLowerCase()
-
-  const account = self.body
-  request.assert.strictEqual(account.name, name,
+  const account = this.body
+  request.assert.strictEqual(account.name, name.toLowerCase(),
     'Account name must match the one in the URL')
+  const result = yield model.setAccount(account, this.req.user)
+  this.body = result.account
+  this.status = result.existed ? 200 : 201
+}
 
-  // SQLite's implementation of upsert does not tell you whether it created the
-  // row or whether it already existed. Since we need to know to return the
-  // correct HTTP status code we unfortunately have to do this in two steps.
-  let existed
-  yield db.transaction(function * (transaction) {
-    existed = yield Account.findByName(name, { transaction })
-    if (existed) {
-      existed.setDataExternal(account)
-      yield existed.save({ transaction })
-    } else {
-      yield Account.createExternal(account, { transaction })
-    }
-  })
-
-  log.debug((existed ? 'updated' : 'created') + ' account name ' + name)
-
-  this.body = this.body.getDataExternal()
-  this.status = existed ? 200 : 201
+module.exports = {
+  getCollection,
+  getConnectors,
+  getResource,
+  putResource
```

Comment:
Why export an object instead of  `exports.putResource = function`? Personally, I prefer exporting an object as well, but I am curious if you have specific reasons for doing so.